src/pages/workspaces/workspace/analysis/file-utils.ts

```diff
@@ -1,113 +1,36 @@
 import _ from 'lodash/fp'
-import { useEffect, useState } from 'react'
 import { Ajax } from 'src/libs/ajax'
-import { useLoadedData } from 'src/libs/ajax/loaded-data/useLoadedData'
-import { reportError, withErrorReporting } from 'src/libs/error'
-import { useCancellation, useStore } from 'src/libs/react-utils'
-import { workspaceStore } from 'src/libs/state'
-import LoadedState from 'src/libs/type-utils/LoadedState'
-import { withHandlers } from 'src/libs/type-utils/lodash-fp-helpers'
 import { NominalType } from 'src/libs/type-utils/type-helpers'
 import * as Utils from 'src/libs/utils'
-import {
-  CloudProviderType,
-  isGoogleWorkspaceInfo,
-  WorkspaceWrapper
-} from 'src/libs/workspace-utils'
+import { GoogleWorkspace } from 'src/libs/workspace-utils'
 import { runtimeTools, ToolLabel, toolToExtensionMap } from 'src/pages/workspaces/workspace/analysis/tool-utils'
 
 
 export type FileName = NominalType<string, 'FileName'> // represents a file with an extension and no path, eg `dir/file.ipynb` =>  `file.ipynb`
 export type AbsolutePath = NominalType<string, 'AbsolutePath'> // represents an absolute path in the context of a cloud storage directory structure, i.e. `dir/file.ipynb`
-export type Extension = NominalType<string, 'Extension'> // represents the substring found after the last dot in a file, eg `dir/file.txt.ipynb` => `ipynb`
+export type FileExtension = 'Rmd' | 'R' | 'ipynb' // represents the substring found after the last dot in a file, eg `dir/file.txt.ipynb` => `ipynb`
```

Comment:
I'm not sure about this. I like the nominal type since it encodes what the value represents. But what enforces that an extension is actually one of these specific values? What prevents me from calling `getExtension('/path/to/file.txt')`? I claim that if TS allows that call, then this type is incorrect.

Reply:
Hmmm... yeah, I'll change this to the original and add another `AnalysisFileExtension` type, which is a more specific version that a few functions interact with